File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineRetakeExam2And3May2019/02.Safari.js
function safari(input) {
    let budget = Number(input[0]);
    let neededFuel = Number(input[1]);
    let dayOfWeek = input[2];

    let gitPrice = 100;
    let fuelPricePerLitter = 2.10;

    let totalNeededMoney = neededFuel * fuelPricePerLitter + gitPrice;

    if(dayOfWeek === 'Saturday') {
        totalNeededMoney = totalNeededMoney * 0.9;
    } else if (dayOfWeek === 'Sunday') {
        totalNeededMoney = totalNeededMoney * 0.8;
    }

    let moneyDiff = Math.abs(budget - totalNeededMoney);

    if(budget >= totalNeededMoney) {
        console.log(`Safari time! Money left: ${moneyDiff.toFixed(2)} lv. `);
    } else {
        console.log(`Not enough money! Money needed: ${moneyDiff.toFixed(2)} lv.`);
    }
}

safari(['1000', '10', 'Sunday'])
safari(['120', '30', 'Saturday'])

File: Programming_Basics_with_Python/programming_basics_online_retake_exam_2_and_3_may_2019/02._safari.py
budget = float(input())
needed_litters_of_fuel = float(input())
day_of_week = input()

price_per_one_litter_fuel = 2.10
price_for_tour_guide = 100
saturday_discount = 0.10
sunday_discount = 0.20

needed_money = needed_litters_of_fuel * price_per_one_litter_fuel + price_for_tour_guide
if day_of_week == 'Saturday':
    needed_money = needed_money - (needed_money * saturday_discount)
elif day_of_week == 'Sunday':
    needed_money = needed_money - (needed_money * sunday_discount)

money_difference = budget - needed_money
if money_difference >= 0:
    print(f'Safari time! Money left: {money_difference:.2f} lv.')
else:
    print(f'Not enough money! Money needed: {abs(money_difference):.2f} lv.')

